akka-stream/src/main/scala/akka/stream/impl/fusing/IteratorInterpreter.scala

```diff
@@ -145,7 +145,8 @@ private[akka] class IteratorInterpreter[I, O](val input: Iterator[I], val ops: S
       outHandlers,
       logics,
       (_, _, _) ⇒ throw new UnsupportedOperationException("IteratorInterpreter does not support asynchronous events."),
-      fuzzingMode = false)
+      fuzzingMode = false,
+      null)
```

Comment:
This field can be a proper non-null value, just use it carefully that was my point. Other places in the interpreter internally use nulls because that class is performance intensive. The reasons are:
 - Option add a new allocation (we eliminated as many object headers as possible, by for example folding two arrays into one: https://github.com/akka/akka/blob/master/akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala#L360) and an extra branch everywhere
 - Adding an explicit "invalid" subclass where interfaces are expected is not always possible, plus might make certain sites megamorphic which weren't before
 - even if any of the above might work with minimal performance impact, they would only throw an exception anyway, which is not really more useful than the NPE since it is an assertion error which the user cannot do anything about (internal bug) and I don't need the exception message because the NPE stacktrace identifies the issue immediately.

That said, if you try to make a different approach that does not use nulls, and does not add a significant benchmark impact then we should take it. But currently the interpreter is capable of being completely inlined by JIT (see https://github.com/akka/akka/issues/19299#issuecomment-181402288) and every branch had been very carefully evaluated in the main loop whether it is truly necessary. I mean matches were replaced by explicitly balanced bitfield based chained ifs (https://github.com/akka/akka/blob/master/akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala#L591), branches were designed so that the common, non-failure path is preferred while the error part contains the extra branches to decode the *exact* error (https://github.com/akka/akka/blob/master/akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala#L346), etc.

The interpreter is the dungeon of streams and it uses patterns that we would not use elsewhere.